superset/assets/javascripts/modules/sandbox.js

```diff
@@ -22,6 +22,10 @@ export default function sandboxedEval(code, context, opts) {
   Object.keys(sandboxContext).forEach(function (key) {
     sandbox[key] = sandboxContext[key];
   });
-  vm.runInNewContext(codeToEval, sandbox, opts);
-  return sandbox[resultKey];
+  try {
+    vm.runInNewContext(codeToEval, sandbox, opts);
+    return sandbox[resultKey];
+  } catch (error) {
+    return () => error;
```

Comment:
Why returning a function that returns an error?

Reply:
The function `sandboxedEval` returns a function (either to mutate the data or generate the tooltip). I'm being consistent here, and returning this ensures that the error is displayed in the tooltip (see second screenshot).